test/object-store/sync/app.cpp

```diff
@@ -2733,14 +2728,10 @@ TEST_CASE("app: sync integration", "[sync][app]") {
         }
         r->commit_transaction();
 
-        // If we haven't gotten an error in more than 5 minutes, then something has gone wrong
-        // and we should fail the test.
-        timed_wait_for(
-            [&] {
-                std::lock_guard<std::mutex> lk(sync_error_mutex);
-                return done;
-            },
-            std::chrono::minutes(5));
+        auto error = wait_for_future(std::move(pf.future), std::chrono::minutes(5)).get();
+        REQUIRE(error.error_code == make_error_code(sync::ProtocolError::limits_exceeded));
+        REQUIRE(error.message == "Sync websocket closed because the server received a message that was too large: "
+                                 "read limited at 16777217 bytes");
```

Comment:
Can you add a check that the server action is indeed a ClientReset?

Reply:
Sure!